### PR TITLE
Feature - [IAM] Delete Policy

### PR DESCRIPTION
**Pull request by bblommers**
Added an additional test for delete_user as well, as that was missing

### CODE REVIEW

File: tests/test_iam/test_iam.py [hunk 1]
```diff
@@ -338,6 +338,16 @@ def test_create_policy():
     response['Policy']['Arn'].should.equal("arn:aws:iam::123456789012:policy/TestCreatePolicy")
 
 
+@mock_iam
+def test_delete_policy():
+    conn = boto3.client('iam', region_name='us-east-1')
+    response = conn.create_policy(PolicyName="TestCreatePolicy", PolicyDocument=MOCK_POLICY)
+    [pol['PolicyName'] for pol in conn.list_policies(Scope='Local')['Policies']].should.equal(['TestCreatePolicy'])
+    #
```
Comment (mikegrima) [inline tests/test_iam/test_iam.py]: Delete this line.

File: tests/test_iam/test_iam.py [hunk 2]
```diff
@@ -713,14 +723,27 @@ def test_mfa_devices():
...
     conn.create_user('my-user')
     conn.delete_user('my-user')
 
 
+@mock_iam()
+def test_delete_user():
+    conn = boto3.client('iam', region_name='us-east-1')
+    with assert_raises(ClientError):
+        conn.delete_user(UserName='my-user')
+    #
```
Comment (mikegrima) [inline tests/test_iam/test_iam.py]: Delete

File: tests/test_iam/test_iam.py [hunk 3]
```diff
@@ -713,14 +723,27 @@ def test_mfa_devices():
...
+    conn.create_user(UserName='my-user')
+    [user['UserName'] for user in conn.list_users()['Users']].should.equal(['my-user'])
+    #
```
Comment (mikegrima) [inline tests/test_iam/test_iam.py]: Delete

**Comment by bblommers** [comment]
Fixed @mikegrima

**Pull request merged**